### DISCUSSION TITLE
Can it be built only with Wayland support?

### DISCUSSION THREAD

**Discussion by fxzxmicah** [VSCodium/vscodium#2072] [Q&A]
`use_ozone=true ozone_auto_platforms=false ozone_platform="wayland" ozone_platform_wayland=true`
For Chromium, these parameters can be used to build. Are there any similar parameters that can be used here?

**Comment by Jeong-ilKajokaya** [comment]
Yo @fxzxmicah 👋

I get what you’re asking — you’re talking about Chromium build flags for Ozone / Wayland, and wondering if VSCodium / Electron has something similar.

---

### The scoop

* VSCodium is basically a patched Electron build, and Electron itself wraps Chromium. So most Chromium build flags don’t directly apply at runtime, but **Electron does respect some environment variables and runtime flags**.

* For Wayland / Ozone, Electron uses the `--ozone-platform` flag or the `ELECTRON_ENABLE_WAYLAND` environment variable:

  ```bash
  # Force Electron to use Wayland
  export ELECTRON_ENABLE_WAYLAND=1
  ./VSCodium

  # Or via command line
  ./VSCodium --ozone-platform=wayland
  ```

* You can also experiment with other Chromium flags Electron supports:

  ```bash
  ./VSCodium --enable-features=UseOzonePlatform --ozone-platform=wayland
  ```

* Note: behavior may differ between Flatpak, Snap, and native builds due to sandboxing or GTK integration.

---

Basically, while you can’t just copy Chromium build args wholesale, these Electron flags are the closest equivalents to get Wayland / Ozone behavior in VSCodium.

**Reply by fxzxmicah** [reply]
I was referring to build parameters, not runtime parameters.

**Reply by Jeong-ilKajokaya** [reply]
Ah, I see! Thanks for clarifying — you meant **build-time parameters** that are set when compiling VSCodium, not the **runtime flags** passed when launching the program.  

That makes sense — I was thinking more about runtime options like `--ozone-platform=wayland`, but I understand now that what you’re asking is about configuring the build itself to include or exclude specific backends.  

![giphy](https://media.giphy.com/media/LllA2dKt1qZuE/giphy.gif?cid=790b76119128c338743cd3de82496952cb6c3fa7a32231c3&ep=v1_user_favorites&rid=giphy.gif&ct=g)
My bad twin
